### DISCUSSION TITLE
[Help] Unsupported Bridge Protocol error

### DISCUSSION THREAD

**Discussion by henjoe** [nanomq/nanomq#1119] [General]
I am already publishing data to my local NanoMQ broker with the topic "data/test/topic1" , I want this data to be forward in my EMQX cloud broker, so as far as I understand it right, this is what "bridging" can do right? 

```
bridges.mqtt {
	nodes = [
		{
			name = emqx
			enable = true
			connector {
				server = "emqx_public_ip:1883"
				proto_ver = 3
				username = user
				password = password
				clean_start = true
				keepalive = 60s
                clientid = "nano-mq-test"
				ssl {
					enable = false
					keyfile = "/etc/certs/key.pem"
					certfile = "/etc/certs/cert.pem"
					cacertfile = "/etc/certs/cacert.pem"
				}
			}
			forwards = ["data/test/topic1"]
			quic_keepalive = 120s
			quic_idle_timeout = 120s
			quic_discon_timeout = 20s
			quic_handshake_timeout = 60s
			hybrid_bridging = true
			congestion_control = cubic
			subscription = [
				{
					topic = "data/#"
					qos = 1
				}
			]
            parallel = 2
            max_send_queue_len = 1024
            max_recv_queue_len = 1024
		}

	]
	sqlite {
		enable = false
		disk_cache_size = 102400
		mounted_file_path = "/tmp/"
		flush_mem_threshold = 100
		resend_interval = 5000
	}

}
```

But when I run my configuration, it gives me the ff error:
![image](https://user-images.githubusercontent.com/49782655/227752193-8b33190a-2dff-44fd-a5e1-74086239e18d.png)

Any help would be highly appreciated.

Thank you.

Regards,
Henjoe

**Comment by JaylinYu** [comment]
server = "emqx_public_ip:1883" 
should be corrected as
server = "mqtt-tcp://emqx_public_ip:1883"

**Comment by henjoe** [comment]
Thanks @JaylinYu  let me try that one.

**Comment by henjoe** [comment]
Hi @JaylinYu 

Bridging works now. However, I still didn't get how "message persistence" works on nanomq. 
What I expect is when nanomq broker (local) is sending data to the cloud (EMQX broker) thru bridging it has automatic offline buffering when a connection is interrupted. Is my understanding correct? 

I have the ff config on my bridge:
![image](https://user-images.githubusercontent.com/49782655/228103910-886688e9-f758-4ab6-a7eb-a1cce58787c3.png)


Again, bridging works but buffering is not. I wish I just need to configure things?

Thanks @JaylinYu  

Regards,
Henjoe

**Comment by wanghaEMQ** [comment]
@henjoe 

Only the msgs with Qos greater than 0 would be buffered.

For a short offline. It's okay to do nothing because NanoMQ would save the msgs to the send queue in memory. The configuration option for the length of send queue is the entry `max_send_queue_len`.

But for a long offline. Enabling SQLite to buffering msgs to external storage is recommended.
